### ISSUE TITLE
Cannot parse EXCEPT keywords in sparksql

### ISSUE THREAD

**Issue by xg1990** [sqlfluff/sqlfluff#3687]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

```sql
SELECT
    * EXCEPT (row_no)
FROM
    table_name
WHERE
    row_no = 1
```
If I run sqlfluff lint on the above sql code, the following error message will appear

```
== [models/curation/test.sql] FAIL                                                                                                                                                                   
L:   2 | P:   6 |  PRS | Line 2, Position 6: Found unparsable section: ' EXCEPT
                       | (row_no)\nFROM\n table_name\nWHE...'
WARNING: Parsing errors found and dialect is set to 'sparksql'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### Expected Behaviour

No Error should appear

### Observed Behaviour

Error happened

### How to reproduce

As mentioned above

### Dialect

sparksql

### Version

sqlfluff, version 1.2.1

Python 3.8.10

### Configuration

N/A

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by xg1990** [comment]
just notice:

`* except()` command is supported by Databricks runtime 9.0+ but not Spark SQL

**Comment by tunetheweb** [comment]
BigQuery has this in it's dialect so could basically just copy it's definition and add some tests:

https://github.com/sqlfluff/sqlfluff/blob/bc147334a8a85b1abdcc007868e7992288247880/src/sqlfluff/dialects/dialect_bigquery.py#L916-L936

> just notice:
> 
> `* except()` command is supported by Databricks runtime 9.0+ but not Spark SQL

We often allow the dialects to be more loose than strictly correct for situations like these. So think it's fine to add.
